allennlp/nn/beam_search.py

```diff
@@ -110,6 +121,11 @@ def search(self,
         # shape: (batch_size, beam_size), (batch_size, beam_size)
         start_top_log_probabilities, start_predicted_classes = \
                 start_class_log_probabilities.topk(self.beam_size)
+        if self.beam_size == 1 and (start_predicted_classes == self._end_index).all():
+            warnings.warn("Empty sequences predicted. You may want to increase the beam size or ensure "
+                          "your step function is working properly.",
+                          RuntimeWarning)
+            return start_predicted_classes.unsqueeze(-1), start_top_log_probabilities
```

Comment:
Should we `expand` the beam dimension here to ensure we agree with the contract we set in the docstring?

Reply:
Hmm maybe I'm not seeing this right but I believe the dimensions returned are correct. I test for this in the test case `test_empty_sequences`.

Reply:
The returned tensors will have shape `(batch_size, beam_size, sequence_length) = (batch_size, 1, 1)` and `(batch_size, beam_size) = (batch_size, 1)`

Reply:
Sorry, got mixed up again. Definitely correct as-is.